force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,11 +138,21 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        Account acc = new Account (Name = accountName);
+        List<Opportunity> oppList = new List<Opportunity>();
+        upsert acc;
+        
+        for (String oppName : oppNames){        // Loop through each Opportunity name
+            Opportunity opp = new Opportunity(
+                Name = oppName,                 // Create an Opportunity with the given name
+                StageName = 'Prospecting',
+                CloseDate = (Date.today()).addDays(7),                   
+                AccountId = acc.Id);    
+            oppList.add(opp);               // Add the Opportunity to the list                 
+        }
+    upsert oppList;
```

Comment:
The tests are passing here, but that seems to be more of a failure of the tests not checking this scenario actually. You should be checking like in question 9 that the account exists and only inserting if it does not. Then doing the same for the opportunities based on the name of the opportunity. Only create them if they do not already exist. Can you try this one again?